guides/release/components/template-lifecycle-dom-and-modifiers.md

```diff
@@ -340,40 +349,71 @@ export default class AudioPlayerComponent extends Component {
   play() {
     // TODO
   }
-}
-```
-
-We'd like to call the `play` method on our `<audio>` element, but how do we get access to that element?
-
-We can give our component access to elements inside of it by using [`ember-ref-modifier`](https://github.com/lifeart/ember-ref-modifier).
-
-> The `{{ref}}` modifier takes a property name and an object, and assigns the modifier's element to that property on the object.
-
-In this case, we'll assign the `<audio>` element to the `audioElement` property in our component:
-
-```handlebars {data-filename="app/components/audio-player.hbs" data-diff="-1,+2"}
-<audio src={{@srcURL}} />
-<audio src={{@srcURL}} {{ref this "audioElement"}} />
 
-<button type="button" {{on "click" this.play}}>Play</button>
-<button type="button">Pause</button>
+  @action
+  pause() {
+    // TODO
+  }
+}
 ```
 
-Now, the component can access the audio element in the `play` method:
+From the component's point of view, all it needs to do is track whether it should be playing or not, since we're going to manage the DOM element with the modifier. So we can just use `@tracked` and a normal property here.
 
-```js {data-filename="app/components/audio-player.js" data-diff="-7,+8"}
+```js {data-filename="app/components/audio-player.js" data-diff="+2,+6,+7,-10,+11,-16,+17"}
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
 
 export default class AudioPlayerComponent extends Component {
+  @tracked isPlaying = false;
+
   @action
   play() {
     // TODO
-    this.audioElement.play();
+    this.isPlaying = true;
+  }
+
+  @action
+  pause() {
+    // TODO
+    this.isPlaying = false;
   }
 }
 ```
 
+Now the component is managing the state of the player, but we still need to translate that into the DOM method calls. So let's create a modifier!
+
+```bash
+ember install ember-modifier
+ember generate modifier play-when
+```
+
+In this modifier, we'll take an argument that specifies whether we should be playing or paused.
+
+```js {data-filename="app/modifiers/play-when.js"}
+import { modifier } from "ember-modifier";
+
+export default modifier((element, [isPlaying]) => {
+  if (isPlaying) {
+    element.play();
+  } else {
+    element.pause();
+  }
+});
+```
+
+We can now attach the modifier to the `audio` element:
+
+```handlebars {data-filename="app/components/audio-player.hbs" data-diff="-1,+2"}
+<audio src={{@srcURL}} />
+<audio src={{@srcURL}} {{play-when this.isPlaying}} />
+
+<button type="button" {{on "click" this.play}}>Play</button>
+<button type="button" {{on "click" this.pause}}>Pause</button>
+```
+
+With that, we're done! Notice that we've separated the concerns of *managing state* from *managing DOM interactions*. The component manages the state. The modifier manages the DOM. And this modifier could be reused for *any* `HTMLMediaElement`, including other `audio` or `video` element in our code.
```

Comment:
```suggestion
With that, we can now click the buttons to play and pause the audio!

In summary, when you want to allow elements in a component to communicate, see if you can separate the concerns of *managing state* and *managing DOM interactions*. The component can manage the state, while a modifier can manage the DOM.

The modifier that we made for the audio player component can be reused on *any* element that implements `play` and `pause` methods. In particular, we can reuse the modifier on an `HTMLMediaElement`, which includes `audio` and `video` elements.
```

Reply:
I *really* liked this; it makes the takeaway much clearer. Thank you!